samples/winapi_drivelist/main.c

```diff
@@ -0,0 +1,58 @@
+#include <hal/debug.h>
+#include <hal/video.h>
+#include <nxdk/mount.h>
+#include <windows.h>
+
+int main(void)
+{
+    XVideoSetMode(640, 480, 32, REFRESH_DEFAULT);
+
+    // Mount some drives for demonstration purposes
+    BOOL ret;
+    ret = nxMountDrive('C', "\\Device\\Harddisk0\\Partition2\\");
+    if (!ret) {
+        debugPrint("Failed to mount C: drive!\n");
+        Sleep(5000);
+        return 1;
+    }
+    ret = nxMountDrive('E', "\\Device\\Harddisk0\\Partition1\\");
+    if (!ret) {
+        debugPrint("Failed to mount E: drive!\n");
+        Sleep(5000);
+        return 1;
+    }
+
+    // Retrieve drive bitmaks. Every bit represents one drive letter
+    DWORD driveBits = GetLogicalDrives();
+    if (driveBits == 0 && GetLastError() != ERROR_SUCCESS) {
+        debugPrint("Failed to retrieve drive bitmask!\n");
+        Sleep(5000);
+        return 1;
+    }
+    debugPrint("Drive bitmask: 0x%x\n\n", driveBits);
+
+
+    // Reserve buffer long enough for all possible drive strings plus null-terminator
+    char buffer[26 * 4 + 1];
+    // IMPORTANT: The size passed to GetLogicalDriveStringsA is WITHOUT the null-terminator, even though it gets written
+    DWORD charsWritten = GetLogicalDriveStringsA(sizeof(buffer)-1, buffer);
+    if (charsWritten == 0) {
```

Comment:
nit: I think you should check for: `charsWritten == sizeof(buffer)-1` to also cover:
> If the buffer is not large enough, the return value is greater than `nBufferLength`.

I'd at least expect a warning in that case.

I know it won't happen here because our buffer is large enough, but this looks like a typical issue a user would make to me.

Reply:
I added a check for `charsWritten > sizeof(buffer) - 1`, which I assume is what you meant? `charsWritten == sizeof(buffer)-1` would only be the case if the buffer gets completely filled, in the sample we get a much smaller value for `charsWritten`.